### PR TITLE
fix(blobs): Validate sidecar `KzgCommitments` and disallow sidecar omission

### PR DESCRIPTION
**Pull request by shotes**
This PR fixes a few things:
- Ensures that the number of commitments in the BeaconBlock and the number of BlobSidecars are equal.
- Ensures that BlobSidecars do not contain duplicate BlobSidecars
- (this implicitly verifies that the KzgCommitments from the BeaconBlock are 1 to 1 with BlobSidecars due to the inclusion proof)

### CODE REVIEW

**Review by calbera — APPROVED** [review]
makes sense to me

File: beacon/blockchain/process_proposal.go [hunk 1]
```diff
@@ -72,56 +78,47 @@ func (s *Service[
...
 
-	// In theory, swapping the order of verification between the sidecars
-	// and the incoming block should not introduce any inconsistencies
-	// in the state on which the sidecar verification depends on (notably
-	// the currently active fork). ProcessProposal should only
-	// keep the state changes as candidates (which is what we do in
-	// VerifyIncomingBlock).
+	// Make sure we have the right number of BlobSidecars
+	numCommitments := len(blk.GetBody().GetBlobKzgCommitments())
+	if numCommitments != sidecars.Len() {
```
Comment (calbera) [inline beacon/blockchain/process_proposal.go:90]: although we already check before calling `.Len()` if `sidecars.IsNil()` I think we should prevent any `sidecars` function from ever panicking on its own

Comment (calbera) [reply beacon/blockchain/process_proposal.go:90]: i.e. here [Len, GetSidecars, Get](https://github.com/berachain/beacon-kit/pull/2291/files#diff-29cacb221d7916f7d9d44e53bc63db6dc41f5e5f816f9634eb5faf5808e98417R38-R48) should all check .IsNil before dereferencing the pointer.

Comment (abi87) [reply beacon/blockchain/process_proposal.go:90]: we do in line 81 right?

**Review by abi87 — APPROVED** [review]
lgtm

**Pull request merged**